crates/rslint_core/src/groups/errors/constructor_super.rs

```diff
@@ -0,0 +1,92 @@
+use crate::rule_prelude::*;
+use ast::{ClassDecl, ClassElement, Expr, Stmt};
+
+declare_lint! {
+    /**
+    Verify calls of `super()` in constructors (constructor-super)
+
+    The `"extends": "rslint:recommended"` property in a configuration file enables this rule.
+
+    Constructors of derived classes must call `super()`. Constructors of non derived classes must not call `super()`. If this is not observed, the JavaScript engine will raise a runtime error.
+
+    This rule checks whether or not there is a valid `super()` call.
+    */
+    #[derive(Default)]
+    ConstructorSuper,
+    errors,
+    tags(Recommended),
+    "constructor-super",
+}
+
+#[typetag::serde]
+impl CstRule for ConstructorSuper {
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        let class_decl = node.try_to::<ClassDecl>()?;
+        let superclass = class_decl.parent();
+        let super_call = class_decl.body()?
+            // work-around for bug where `.body()?.elements()` returns only one
+            // element for whatever reason
+            .syntax().children().filter_map(|x| x.try_to::<ClassElement>())
+            // get the first constructor we can find. there should only be one.
+            .find_map(|x| match x {
+                ClassElement::Constructor(c) => Some(c),
+                _ => None,
+            })?
+            // get only the expression statements for the constructor's body
+            .body()?
+            .stmts()
+            .filter_map(|x| match x {
+                Stmt::ExprStmt(expr) => expr.expr(),
+                _ => None,
+            })
+            // find a super call expression. there should only be one.
+            .find(|x| matches!(x, Expr::SuperCall(_)));
+        
+        // if it's a subclass xor it's constructor calls super, show error.
+        match (superclass, super_call) {
+            (Some(class), None) => {
+                let diagnostic = ctx
+                    .err(
+                        self.name(),
+                        "constructor of derived class must call super",
+                    )
+                    .primary(
+                        class.syntax(),
+                        "superclass specified here, but super was not called",
+                    );
```

Comment:
I don't think this error is structured well. Id rather have the primary label point to the entire constructor, then have a secondary label which points to the superclass. Since the primary label should point the user to directly where the problem is, while secondary labels should add more context/point things out.

Reply:
The entire constructor? From beginning to ending brace?

Reply:
Yes the entire thing from the constructor keyword to the end of the body

Reply:
What should the primary label to the constructor say? Is "no super call found within constructor" okay?

Reply:
I'll just go with that for now.